tests/Carbon/IsTest.php

```diff
@@ -708,28 +708,47 @@ public function testIsSaturday()
     public function testIsStartOfDay()
     {
         $this->assertTrue(Carbon::parse('00:00:00')->isStartOfDay());
+        $this->assertTrue(Carbon::parse('00:00:00.999999')->isStartOfDay());
         $this->assertTrue(Carbon::now()->startOfDay()->isStartOfDay());
 
         $this->assertFalse(Carbon::parse('15:30:45')->isStartOfDay());
         $this->assertFalse(Carbon::now()->endOfDay()->isStartOfDay());
     }
 
-    public function testIsMidnight()
+    public function testIsStartOfDayWithMicroseconds()
     {
-        $this->assertTrue(Carbon::parse('00:00:00')->isMidnight());
+        $this->assertTrue(Carbon::parse('00:00:00')->isStartOfDay(true));
+        $this->assertTrue(Carbon::now()->startOfDay()->isStartOfDay(true));
 
-        $this->assertFalse(Carbon::parse('15:30:45')->isMidnight());
+        $this->assertFalse(Carbon::parse('00:00:00.000001')->isStartOfDay(true));
     }
 
     public function testIsEndOfDay()
     {
-        $this->assertTrue(Carbon::parse('23:59:59.999999')->isEndOfDay());
+        $this->assertTrue(Carbon::parse('23:59:59')->isEndOfDay());
+        $this->assertTrue(Carbon::parse('23:59:59.000000')->isEndOfDay());
         $this->assertTrue(Carbon::now()->endOfDay()->isEndOfDay());
 
         $this->assertFalse(Carbon::parse('15:30:45')->isEndOfDay());
         $this->assertFalse(Carbon::now()->startOfDay()->isEndOfDay());
     }
 
+    public function testIsEndOfDayWithMicroseconds()
+    {
+        $this->assertTrue(Carbon::parse('23:59:59.999999')->isEndOfDay(true));
+        $this->assertTrue(Carbon::now()->endOfDay()->isEndOfDay(true));
```

Comment:
I guess it will fail on PHP 7.0, but we can just skip the test with a reference to the bug.